### ISSUE TITLE
Crash in BugsnagClient.m:827 

### ISSUE THREAD

**Issue by sethfri** [bugsnag/bugsnag-cocoa#941]
### Describe the bug
Crash of unknown origin coming from Bugsnag. We have seen 4 crashes coming from 4 users so far.

### Steps to reproduce
Unknown

### Environment
* Bugsnag version: 6.2.5
* CocoaPods version: 1.9.3
* Carthage version (if any): n/a
* iOS/tvOS/macOS version(s): 14.2.0
* Simulator or physical device: iPhone 7 Plus
* Xcode version: 12.1
* Swift version (if applicable): 5.1

### Stack Trace

<details><summary>Stack trace:</summary>

```
CrashReporter Key:  e8e0e23387e547f66b7a44fcd94779533c62df80
Version:            3.28
Role:               Background
OS Version:         iOS 14.2
Exception Type:     EXC_BAD_ACCESS 

EXC_BAD_ACCESS: Attempted to dereference garbage pointer 0x800000001.

0  libobjc.A.dylib         _objc_msgSend
1  MyApp                    __42-[BugsnagClient setupConnectivityListener]_block_invoke (BugsnagClient.m:827:9)
2  SystemConfiguration     _reachPerformAndUnlock
3  SystemConfiguration     _____SCNetworkReachabilityRestartResolver_block_invoke_2
4  libnetwork.dylib        ___nw_resolver_update_client_block_invoke
5  libdispatch.dylib       __dispatch_call_block_and_release
6  libdispatch.dylib       __dispatch_client_callout
7  libdispatch.dylib       __dispatch_lane_serial_drain$VARIANT$mp
8  libdispatch.dylib       __dispatch_lane_invoke$VARIANT$mp
9  libdispatch.dylib       __dispatch_workloop_worker_thread
10 libsystem_pthread.dylib __pthread_wqthread
```
</details>

**Comment by nickdowell** [comment]
Hi @sethfri thanks for reporting this.

This looks like the same underlying issue as #900 which was fixed in [release v6.2.6](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.2.6)

**Comment by sethfri** [comment]
Awesome! Closing this down then
